src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/OrKeyFrameCriteriaEvaluator.cs

```diff
@@ -5,13 +5,14 @@ namespace RegressionGames.StateRecorder.BotSegments
 {
     public static class OrKeyFrameCriteriaEvaluator
     {
-        public static bool Matched(bool firstSegment, int segmentNumber, KeyFrameCriteria criteria)
+        public static bool Matched(bool firstSegment, BotSegment botSegment, KeyFrameCriteria criteria)
```

Comment:
I had intentionally not passed the full BotSegment reference in this methods to limit data access and avoid the potential ambiguity in introduces.  By passing this, implements can now access the entire raw key frame criteria from the segment instead of being limited to the one criteria being passed.  Let's undo this part of the changes please.